.github/workflows/deploy.yaml

```diff
@@ -50,6 +50,7 @@ jobs:
 
       - uses: pypa/gh-action-pypi-publish@v1.5.0
         name: Deploy to Test PyPi
+        if: github.event_name == 'push'
```

Comment:
I would also just push to test-pypi if on main branch:
```
github.ref == 'refs/heads/main'
```

Reply:
Since there is a separate stage for wheel building, I think it is very unlikely that the test-pypi stage will fail because of changes on the code-base. We only need to assure, that wheels are built. I see the deploy to test-pypi as something like "nightly builds" and they should come from main. Also the test-pypi stage is there to assure it is working before a release is made. And since releases come from tags made on main, it should just work there. From my point of view, there is no need to push to test-pypi in any other situation.

Reply:
OK, good points. I've updated to only deploy on main.